game-core/src/main/java/games/strategy/triplea/delegate/battle/firinggroups/FiringGroup.java

```diff
@@ -0,0 +1,80 @@
+package games.strategy.triplea.delegate.battle.firinggroups;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.UnitAttachment;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import lombok.NonNull;
+import lombok.Value;
+
+/**
+ * Breaks list of units into groups of non suicide on hit units and each type of suicide on hit
+ * units since each type of suicide on hit units need to roll separately to know which ones get
+ * hits.
+ */
+@Value(staticConstructor = "of")
+public class FiringGroup {
+  private final @NonNull Collection<Unit> firingGroup;
+  private final @NonNull Collection<Unit> validTargets;
+  private final boolean suicide;
+  // type is currently only used by Aa firing groups
```

Comment:
Should it be an enum? If not, can the comment document possible values at least?

Reply:
As far as I can tell, this comes out of the game xml.  It comes out of UnitAttachment#getAllOfTypeAas.  So I think the possible values are user defined.

Reply:
I also wondered if an enum made sense. We'd need a mapping of string values to enum values so the two could be mapped (and leave allowance for an 'other' category). Ultimately though I think that is an update too far, we should be focused on just moving code for this iteration. The clean-ups and changes are probably even a step too far for this iteration (but those are already started, so we should finish; to be clear, simply moving code is probably a good first PR, then changing, cleaning up is a good follow up. We do run into a conflict that is endemic in TripleA where you simply need far too many iterations to make all changes bite-sized. Strategic layering of commits can help)

Reply:
I don't think it can be enum.  This is a way to group the aa units in a map.  A map could have one aa group or it could have 30 aa groups.  This name is also presented to the user in the battle dialog when each group fires.

What could be done is to grab the group name in FireAa after the FiringGroups have been established instead of here during the creation of the FiringGroups.